mirrord-config/src/env.rs

```diff
@@ -8,17 +8,56 @@ use crate::{
     util::{MirrordToggleableConfig, VecOrSingle},
 };
 
+/// Allows the user to set or override a local process' environment variables with the ones from a
+/// remote pod.
+///
+/// Which environment variables to load from the remote are controlled by setting either `include`
+/// or `exclude`.
+///
+/// ## Examples
+///
+/// - Include every environment variable from the remote pod (default).:
+///
+/// ```yaml
+/// # mirrord-config.yaml
+///
+/// include = "*"
+/// ```
+///
+/// Some environment variables are excluded by default (`PATH` for example), including these
+/// requires specifying them with `include`, see [`mirrord_agent::env::EnvFilter`].
+///
+/// - Include the remote pod's environment variables "project", "database":
+///
+/// ```yaml
+/// # mirrord-config.yaml
+///
+/// include = "project;database"
+/// ```
+///
+/// - Exclude the remote pod's environment variables "project", "database", and include everything
+///   else:
+///
+/// ```yaml
+/// # mirrord-config.yaml
+///
+/// exclude = "project;database"
+/// ```
 #[derive(MirrordConfig, Default, Deserialize, PartialEq, Eq, Clone, Debug)]
 #[cfg_attr(feature = "schema", derive(schemars::JsonSchema))]
 #[serde(deny_unknown_fields)]
 #[config(map_to = EnvConfig)]
 pub struct EnvFileConfig {
+    /// Include only these remote environment variables in the local process.
     #[config(env = "MIRRORD_OVERRIDE_ENV_VARS_INCLUDE")]
     pub include: Option<VecOrSingle<String>>,
 
+    /// Include the remote environment variables in the local process that are **NOT** specified by
+    /// this option.
     #[config(env = "MIRRORD_OVERRIDE_ENV_VARS_EXCLUDE")]
     pub exclude: Option<VecOrSingle<String>>,
 
+    // TODO(alex) [mid] 2022-10-29: Have no idea about this.
```

Comment:
This is a replacement for `.env` file or setting your own env then adding exclude.
For example, you want to set `AWS_REGION=1` but the remote pod has `AWS_REGION=2`
To have `AWS_REGION=1` applied, you need to set it at the local process + set it to exclude.
This is a shortcut to do it in one spot.